sdk/src/main/java/com/horizen/node/NodeWallet.java

```diff
@@ -33,6 +33,8 @@ public interface NodeWallet {
 
     <S extends Secret> List<S> secretsByProposition(ProofOfKnowledgeProposition<S> proposition);
 
+    <S extends Secret> Optional<S> secretByProposition(byte[] proposition);
```

Comment:
let's call it `secretByPublicKeyBytes(byte[] publicKey)`
since this method is not about all propositions, but only for the case of PrivateKey <-> PublicKey relations.
For custom proposition, for example, with multiple public keys inside (K-of-N multisig) this method has no sense.

Reply:
Yes better, done.